react/features/virtual-background/components/VirtualBackgroundDialog.js

```diff
@@ -105,62 +125,90 @@ function VirtualBackground({ _jitsiTrack, _selectedThumbnail, dispatch, t }: Pro
             jitsiLocalStorage.setItem('virtualBackgrounds', JSON.stringify(storedImages));
         } catch (err) {
             // Preventing localStorage QUOTA_EXCEEDED_ERR
-            err && deleteStoredImage(storedImages[0]);
+            err && setStoredImages(storedImages.slice(1));
         }
         if (storedImages.length === backgroundsLimit) {
             setStoredImages(storedImages.slice(1));
         }
     }, [ storedImages ]);
 
-    const [ selected, setSelected ] = useState('');
 
     const enableBlur = useCallback(async () => {
-        isloading(true);
-        setSelected('blur');
-        await dispatch(setVirtualBackground('', false));
-        await dispatch(toggleBackgroundEffect(true));
-        isloading(false);
-    }, [ dispatch ]);
+        setOptions({
+            backgroundType: 'blur',
+            enabled: true,
+            blurValue: 25,
+            selectedThumbnail: 'blur'
+        });
+    }, []);
 
-    const removeBackground = useCallback(async () => {
-        isloading(true);
-        setSelected('none');
-        await dispatch(setVirtualBackground('', false));
-        await dispatch(toggleBackgroundEffect(false));
-        isloading(false);
-    }, [ dispatch ]);
+    const enableBlurKeyPress = useCallback(e => {
+        if (e.key === ' ' || e.key === 'Enter') {
+            e.preventDefault();
+            enableBlur();
+        }
+    }, [ enableBlur ]);
 
-    const setImageBackground = useCallback(async e => {
-        const imageId = e.currentTarget.getAttribute('data-imageid');
-        const image = images.find(img => img.id === imageId);
+    const enableSlideBlur = useCallback(async () => {
+        setOptions({
+            backgroundType: 'blur',
+            enabled: true,
+            blurValue: 8,
+            selectedThumbnail: 'slight-blur'
+        });
+    }, []);
 
-        if (image) {
-            isloading(true);
-            setSelected(image.id);
-            await dispatch(setVirtualBackground(await toDataURL(image.src), true));
-            await dispatch(toggleBackgroundEffect(true));
-            isloading(false);
+    const enableSlideBlurKeyPress = useCallback(e => {
+        if (e.key === ' ' || e.key === 'Enter') {
+            e.preventDefault();
+            enableSlideBlur();
         }
-    }, [ dispatch ]);
+    }, [ enableBlur ]);
+
+    const removeBackground = useCallback(async () => {
+        setOptions({
+            enabled: false,
+            selectedThumbnail: 'none'
+        });
+    }, []);
+
+    const removeBackgroundKeyPress = useCallback(e => {
```

Comment:
These callbacks look very similar. I think a custom hook  could be created to avoid repetition.